src/components/tl/ucp/tl_ucp_coll.h

```diff
@@ -121,6 +126,23 @@ typedef struct ucc_tl_ucp_task {
             ucc_ee_executor_task_t *etask;
             ucc_ee_executor_t      *executor;
         } allreduce_kn;
+        struct {
+            int                                        reduce_in_progress;
+            ucp_rkey_h                                *src_rkeys; //unpacked
+            ucp_rkey_h                                *dst_rkeys; //unpacked
+            void                                     **sbufs;
+            void                                     **rbufs;
+            ucc_tl_ucp_allreduce_sw_pipeline          *pipe;
+            ucc_ee_executor_task_t                    *etask;
+            ucc_ee_executor_t                         *executor;
+            int                                        put_window_size;
+            int                                        num_get_bufs;
```

Comment:
no need to duplicate put_window_size and num_get_bufs here, they are part of team config